src/presentation/http/router/note.test.ts

```diff
@@ -518,6 +518,204 @@ describe('Note API', () => {
+        creatorId: user.id,
+      });
+
+      /** Create test note - a child note */
+      const childNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: childNote.id,
+        isPublic: true,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: parentNote.id,
+        noteId: childNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: parentNote.publicId,
+            content: parentNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns two note parents in case when the note parents IDs relations are provided in a different order than expected', async () => {
+      /** Create test user */
+      const user = await global.db.insertUser();
+
+      /** Create access token for the user */
+      const accessToken = global.auth(user.id);
+
+      /** Create test note - a grand parent note */
+      const firstNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note - a parent note */
+      const secondNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note - a child note */
+      const thirdNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: secondNote.id,
+        isPublic: true,
+      });
+
+      /** Create note relation between parent and grandParentNote */
+      await global.db.insertNoteRelation({
+        parentId: firstNote.id,
+        noteId: thirdNote.id,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: thirdNote.id,
+        noteId: secondNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${secondNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: firstNote.publicId,
+            content: firstNote.content,
+          },
+          {
+            id: thirdNote.publicId,
+            content: thirdNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns one parent note when the user is not in the parent note\'s team but has note relation', async () => {
```

Comment:
case is not clear for me. Which part of code are you testing?

Reply:
This test case when the notes are not created by the same user

Reply:
I don't see any special logic in a code for that case. Do we need to test it if we don't have a corresponded statements in a code?

Reply:
It doesn't check anything special, as the note parents get retrieved through note relation. I'll remove it